src/benchmark.cc

```diff
@@ -348,21 +348,44 @@ void RunBenchmarks(const std::vector<BenchmarkInstance>& benchmarks,
 
     size_t num_repetitions_total = 0;
 
+    // This perfcounters object needs to be created before the runners vector
+    // below so it outlasts their lifetime.
+    PerfCountersMeasurement perfcounters(
+        StrSplit(FLAGS_benchmark_perf_counters, ','));
+
+    // Vector of benchmarks to run
     std::vector<internal::BenchmarkRunner> runners;
     runners.reserve(benchmarks.size());
+
+    // Count the number of benchmarks with threads to warn the user in case
```

Comment:
nit: I think this is number of threads that will run benchmarks. The benchmarks could have threads as part of their internal implementation otherwise no problem.

Reply:
That is true. However I don't think we have control over that, or even if we could possibly know how many threads a process had. That said, I don't think it matters if you disclose that the usage will be aggregated per process and test.

Reply:
Maybe I am not understanding what you are saying?

Reply:
i think the wording is correct. this is the number of benchmarks that have requested to run threaded.